### PR TITLE
Fix issue whereby a live stream update was closing openable cards.

### PR DESCRIPTION
**Pull request by i-like-robots**
- Uses `shouldComponentUpdate` to only re-render openable cards when necessary (perf++)
- Doesn't re-load article if you've already opened it
- Applies a template property to each stream item so that keys can be generated consistently (fixes https://jira.ft.com/browse/NFT-318)
- Also fixes https://jira.ft.com/browse/NFT-316 by switching from `window.scrollY` to `window. pageYOffset`

### CODE REVIEW

File: components/card-open-in-stream/inline-article/index.js [hunk 1]
```diff
@@ -15,46 +15,54 @@ module.exports = React.createClass({
 	},
 
 	handleClick () {
-		if (!this.state.articleExpanded) {
+		if (this.state.articleLoading) {
+			return;
```
Comment (andygnewman) [inline components/card-open-in-stream/inline-article/index.js]: button is disabled at this point so should be impossible for this to happen

File: components/card-open-in-stream/inline-article/index.js [hunk 2]
```diff
@@ -15,46 +15,54 @@ module.exports = React.createClass({
...
+				this.fetchArticle();
+			}
 		}
 	},
 
 	articleClosing () {
 		this.setState({
 			articleExpanded: false,
-			buttonLabel: 'Show more',
-			inlineArticle: null
```
Comment (andygnewman) [inline components/card-open-in-stream/inline-article/index.js]: Presume this is to retain the article, so don't need to fetch it again if reopened

File: components/card-open-in-stream/inline-article/index.js [hunk 3]
```diff
@@ -64,10 +72,10 @@ module.exports = React.createClass({
 			buttonLabel: 'Show less',
 			inlineArticle: html
 		});
-		const originalPosition = window.scrollY;
```
Comment (andygnewman) [inline components/card-open-in-stream/inline-article/index.js]: This and the scrollTo on line 69 were to overcome the scroll action that the focus would undertake.

When you reinstate it you could make the window.scrollY IE compatible document.documentElement.scrollTop.

File: components/card-open-in-stream/inline-article/index.js (hunk 1 above)
Comment (i-like-robots) [inline components/card-open-in-stream/inline-article/index.js]: I was just being belt and braces to cover all states. I think it is actually possible to click twice (state is set async) but this wouldn't save us from that anyway.

File: components/card-open-in-stream/inline-article/index.js (hunk 2 above)
Comment (i-like-robots) [inline components/card-open-in-stream/inline-article/index.js]: Yep!

File: components/card-open-in-stream/inline-article/index.js (hunk 3 above)
Comment (i-like-robots) [inline components/card-open-in-stream/inline-article/index.js]: window.pageYoffset FTW

**Comment by andygnewman** [comment]
:+1:

**Pull request merged**